src/Semantic/Api/Diffs.hs

```diff
@@ -69,12 +71,19 @@ jsonError :: Applicative m => BlobPair -> SomeException -> m (Rendering.JSON.JSO
 jsonError blobPair (SomeException e) = pure $ renderJSONDiffError blobPair (show e)
 
 diffGraph :: (Traversable t, DiffEffects sig m) => t BlobPair -> m DiffTreeGraphResponse
-diffGraph blobs = DiffTreeGraphResponse . V.fromList . toList <$> distributeFor blobs go
+diffGraph blobs = do
+  graph <- distributeFor blobs go
+  pure $ defMessage & P.files .~ toList graph
   where
     go :: DiffEffects sig m => BlobPair -> m DiffTreeFileGraph
     go blobPair = diffWith jsonGraphDiffParsers (pure . jsonGraphDiff blobPair) blobPair
       `catchError` \(SomeException e) ->
-        pure (DiffTreeFileGraph path lang mempty mempty (V.fromList [ParseError (T.pack (show e))]))
+        pure $ defMessage
+          & P.path .~ path
+          & P.language .~ lang
+          & P.vertices .~ mempty
+          & P.edges .~ mempty
```

Comment:
Is it generally the case that these fields would be default-initialized to the same value as `mempty`?

Reply:
Yes, we could remove the setting to `mempty` off entirely.